### ISSUE TITLE
xfdashboard appears in budgie's panel

### ISSUE THREAD

**Issue by fossfreedom** [gmc-holle/xfdashboard#121]
Quick question

running the very latest .91 series (i.e. no longer running the .90 series with the patches you requested previously I test)

running xfdashboard -d &

then I manually run xfdashboard

issue1:
I see the xfdashboard popup in the budgie panel - I'm running budgie with the hints patch as suggested previously.

![Imgur](http://i.imgur.com/sbFg9OT.png)

Press esc to close dashboard

I still see xfdashboard in the panel

issue2:
Running xfdashboard manually again.  xfdashboard doesnt come to the foreground.  I can however click on the panel icon and xfdashboard appears.  Pressing Esc closes xfdashboard - but the panel icon remains. 

Thoughts?

**Comment by gmc-holle** [comment]
Sorry, I have no idea at the moment and big problems to reproduce it. Maybe I get it reproduced in my VM running budgie.

**Comment by fossfreedom** [comment]
I'm just cutting beta 2 of budgie today/tomorrow.  I'll let you know when I've released this and you can try through this way if you consider this an easier way


Sent from Outlook Mobile






On Mon, Mar 28, 2016 at 11:12 PM -0700, "gmc-holle" <notifications@github.com> wrote:





Sorry, I have no idea at the moment and big problems to reproduce it. Maybe I get it reproduced in my VM running budgie.

---

**Comment by fossfreedom** [comment]
Hi - sorry for the delay - I've just managed to upload beta 2.  Sourceforge is up and down like a yo-yo at the moment!

 - https://sourceforge.net/projects/budgie-remix/files/?source=navbar

**Comment by fossfreedom** [comment]
just trying the new version that has been released.

Same issue as above.  Also seeing the following error messages in a terminal:

(xfdashboard:2251): xfdashboard-CRITICAL **: xfdashboard_window_tracker_monitor_get_number: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(self)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: _xfdashboard_windows_view_on_window_monitor_changed: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(inNewMonitor)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: xfdashboard_window_tracker_monitor_get_number: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(self)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: _xfdashboard_windows_view_on_window_monitor_changed: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(inOldMonitor)' failed
dad@dad-VirtualBox:~$ 
(xfdashboard:2251): xfdashboard-CRITICAL **: xfdashboard_window_tracker_monitor_get_number: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(self)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: _xfdashboard_windows_view_on_window_monitor_changed: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(inNewMonitor)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: xfdashboard_window_tracker_monitor_get_number: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(self)' failed

(xfdashboard:2251): xfdashboard-CRITICAL **: _xfdashboard_windows_view_on_window_monitor_changed: assertion 'XFDASHBOARD_IS_WINDOW_TRACKER_MONITOR(inOldMonitor)' failed

**Comment by gmc-holle** [comment]
I tried your last budgie remix iso in a VM and I'm pretty sure that xfdashboard is behaving the right way.

I started xfdashboard as daemon and the window icon appeared in the panel even if the window was not visible on screen. Then I check the window properties with `xwininfo` and the correct window hints are set (skip-pager and skip-taskbar). So either you are using the unpatched budgie wm and/or applets or the applet does not recognize changing window hints - skip-pager and skip-taskbar are set later on first display of the window. But even if I let the window of xfdashboard appear and disappear again the applet still shows window icon for xfdashboard.

Here is the output of `xwininfo`:

```
~ $ cat xfdashboard-wininfo-running_on_budgie_remix-20160401_1331.txt 

xwininfo: Window id: 0x2a00018 "xfdashboard"

  Root window id: 0x25d (the root window) (has no name)
  Parent window id: 0x140188b (has no name)
     0 children.

  Absolute upper-left X:  0
  Absolute upper-left Y:  0
  Relative upper-left X:  0
  Relative upper-left Y:  0
  Width: 1280
  Height: 720
  Depth: 32
  Visual: 0x41
  Visual Class: TrueColor
  Border width: 0
  Class: InputOutput
  Colormap: 0x2a00017 (not installed)
  Bit Gravity State: ForgetGravity
  Window Gravity State: NorthWestGravity
  Backing Store State: NotUseful
  Save Under State: no
  Map State: IsViewable
  Override Redirect State: no
  Corners:  +0+0  -0+0  -0-0  +0-0
  -geometry 1280x720+0+0

  Bit gravity: ForgetGravity
  Window gravity: NorthWestGravity
  Backing-store hint: NotUseful
  Backing-planes to be preserved: 0xffffffff
  Backing pixel: 0
  Save-unders: No

  Someone wants these events:
      KeyPress
      KeyRelease
      ButtonPress
      ButtonRelease
      EnterWindow
      LeaveWindow
      PointerMotion
      Exposure
      StructureNotify
      FocusChange
      PropertyChange
  Do not propagate these events:
  Override redirection?: No

  Window manager hints:
      Client accepts input or input focus: Yes
      Initial state is Normal State
      Displayed on desktop 0
      Window state:
          Skip Pager
          Skip Taskbar
          Fullscreen
          Hidden
          Above
          Sticky
      Process id: 2128 on host budgie-remix

  Normal window size hints:
  No zoom window size hints defined

  No window shape defined
  No border shape defined
```

You see at `Window state:` that both skip-pager and skip-taskbar are set. So the window should not be shown in the applet and in pager when ALT+TAB is pressed. I wasn't able to test the pager in the VM as ALT+TAB is grabbed by my window manager (xfwm4) and never reaches the VM.

At the moment I have to  classify this issue as **invalid**

**Comment by fossfreedom** [comment]
any chance I can talk this through with you? google+ hangout? https://plus.google.com/u/0/+davidfossfreedommohammed/posts

I'm not so worried about the dashboard icon appearing in the panel.  Its when you run xfdashboard manually for the second time - the dashboard doesnt appear.  It appears the dashboard is minimised to the panel.  So when you click the icon in the panel - the dashboard appears.

**Comment by fossfreedom** [comment]
alternatively I've sent an invite to the budgie-remix development chat room.

**Comment by fossfreedom** [comment]
@ikeydoherty

re "So either you are using the unpatched budgie wm and/or applets or the applet does not recognize changing window hints - skip-pager and skip-taskbar are set later on first display of the window. "

As far as I know v10.2.5 of budgie now includes those "hints" and shouldnt display the window in the panel?

**Comment by fossfreedom** [comment]
@gmc-holle - when its in its "icon - please click me to maximise mode" I've run this which doesnt look like your trace:

<pre>
dad@dad-VirtualBox:~$ xwininfo -name "xfdashboard" -all

xwininfo: Window id: 0x2c00001 "xfdashboard"

  Root window id: 0x26c (the root window) (has no name)
  Parent window id: 0x26c (the root window) (has no name)
     0 children.

  Absolute upper-left X:  10
  Absolute upper-left Y:  10
  Relative upper-left X:  10
  Relative upper-left Y:  10
  Width: 10
  Height: 10
  Depth: 0
  Visual: 0x19b
  Visual Class: TrueColor
  Border width: 0
  Class: InputOnly
  Colormap: 0x0 (not installed)
  Bit Gravity State: ForgetGravity
  Window Gravity State: NorthWestGravity
  Backing Store State: NotUseful
  Save Under State: no
  Map State: IsUnMapped
  Override Redirect State: no
  Corners:  +10+10  -942+10  -942-548  +10-548
  -geometry 10x10+10+10

  Bit gravity: ForgetGravity
  Window gravity: NorthWestGravity
  Backing-store hint: NotUseful
  Backing-planes to be preserved: 0xffffffff
  Backing pixel: 0
  Save-unders: No

  Someone wants these events:
      Exposure
      VisibilityChange
      StructureNotify
      PropertyChange
  Do not propagate these events:
  Override redirection?: No

  No window manager hints defined
  Window manager hints:
      Process id: 3291 on host dad-VirtualBox

  Normal window size hints:
      Program supplied size: 10 by 10
  No zoom window size hints defined

  No window shape defined
  No border shape defined
</pre>

**Comment by gmc-holle** [comment]
Sorry for the delay but currently there's lot of work and I prepare for vacation :)

I have tried what you did but cannot reproduce it in budgie-remix. I started xfdashboard in daemon mode (`xfdashboard -d`) from a terminal and no icon was displayed in the task bar applet. That's right as no window was opened and mapped so far. `xwininfo` (started in another terminal) does "find" a window which is in unmapped state. **This** matches your output above and so far everything is as I expected.

Now I call `xfdashboard` from the other terminal (with and without `-t` parameter - it doen't matter) and quitted it by pressing ESC. Now the situation has changed and the output of `xwininfo` matches the output I posted in the previous comment. The window was mapped when I run `xfdashboard` in the terminal but hidden (see Window state) when xfdashboard was "quitted". In addition the window states "Skip Pager" and "Skip Taskbar" are set now. These both window states can only be set after the window was really created.

I may be wrong but I expect the following to happen:

1.) The "hidden" window state is set and that means a previous visible window is now minimized. The window list applet can still show this window list to reselect it.
2.) The "skip taskbar" window state should be respected by the window list applet and shouldn't show the window in its list regardless if the window is visible or hidden/minimized.
3.) The "skip pager" window state should prevent the window to be listed in the window manager's pager which is shown when pressing ALT+TAB (the usual key combination).

The first point happens and is totally ok in my opinion but the applet does not respect the second one and that's an error in my opinion. If I remove the "Icon task list" applet from the panel and add "Task list" applet to the panel it behaves as expected.

Regarding the second problem that xfdashboard's window does not re-appear when called simply with `xfdashboard` or with `xfdashboard -t`. That's strange but may be a window manager issue. I saw a lot of windows focus problems,e .g. when starting `d-feet` from a terminal window the application window of d-feet is displayed below the terminal window where I started d-feet from. The application window of d-feet is neither raised to front nor focused when opened. I don't know if this behaviour is wanted and I also don't know if this may prevent xfdashboard's window to re-appear. The only thing I can say is that "my" window appeared when I killed the window manager process. That means the window is still available and was visible at the time I killed window manager.

What can I do? I'm afraid to say ... nothing :(
